lib/vanagon/component/source/git.rb

```diff
@@ -26,14 +26,39 @@ class << self
           #        git command has failed. Useful in instances where a URL
           #        prompts for credentials despite not being a git remote
           # @return [Boolean] whether #url is a valid Git repo or not
+
+          # [RE-13837] This ought to be the way to do this. Unfortunately,
+          # there's a bug in Git.ls_remote that when ssh prints something like
+          #  Warning: Permanently added 'github.com,192.30.255.113' (RSA)
+          # Git.ls_remote attempts to parse it as actual git output and fails
+          # with: NoMethodError: undefined method `split' for nil:NilClass
+          #
+          # We'll work around that case by calling 'git ls-remote' directly ourselves.
+          #
+          # I'm leaving in the broken version here for a time when the ruby-git library
+          # is fixed.
+
+          #def valid_remote?(url, timeout = 0)
+          #  Timeout.timeout(timeout) do
+          #    !!::Git.ls_remote(url)
+          #  end
+          #rescue ::Git::GitExecuteError
+          #  false
+          #rescue Timeout::Error
+          #  false
+          #end
+
           def valid_remote?(url, timeout = 0)
             Timeout.timeout(timeout) do
-              !!::Git.ls_remote(url)
+              Vanagon::Utilities.local_command("git ls-remote #{url} > /dev/null 2>&1")
+              return false unless $?.exitstatus.zero?
+              return true
             end
-          rescue ::Git::GitExecuteError
-            false
+          rescue RuntimeError # rubocop:disable Lint/ShadowedException
+            ## rubocop is right by flagging this but I think we need it in this case.
```

Comment:
Is there a reason the runtime error needs to come before the timeout error?

Reply:
In short, no. What happened was the `::Git::GitExecuteError` from `Git.ls_remote` with `Vanagon::Utilities.local_command`. I swapped the rescue and the latter throws `RuntimeError` on error.

I see the value in moving it after `rescue Timeout::Error`

Reply:
yeah, and then we can get rid of the rubocop exception :)